client/src/components/Sidebar/ChatContent.js

```diff
@@ -32,10 +57,23 @@ const ChatContent = ({ conversation }) => {
         <Typography className={classes.username}>
           {otherUser.username}
         </Typography>
-        <Typography className={classes.previewText}>
+        <Typography
+          className={
+            unreadMessages > 0 && messages[messages.length - 1].senderId !== user.id
+            ? classes.previewUnreadText
+            : classes.previewText
+          }
+        >
           {latestMessageText}
         </Typography>
       </Box>
+      {unreadMessages > 0 && messages[messages.length - 1].senderId !== user.id &&
+        <Box className={classes.notification}>
+          <Typography className={classes.unreadCount}>
+            {unreadMessages}
+          </Typography>
+        </Box>
+      }
```

Comment:
I noticed a bug when removing the first condition `unreadMessages > 0` and using the badge component when you create a new conversation. Initially, the notification bubble will appear with `NaN` because the `unreadMessages` count is undefined. One way we could fix this is by setting the property to 0 when creating fake conversations, but I chose to reinsert the check back into the conditional render and that solved the issue.